services/workflows-service/src/business/business.service.ts

```diff
@@ -16,16 +16,74 @@ import { plainToClass } from 'class-transformer';
 import dayjs from 'dayjs';
 import { lastValueFrom } from 'rxjs';
 import { BusinessRepository } from './business.repository';
+import { CustomerService } from '@/customer/customer.service';
+import { UnifiedApiClient } from '@/common/utils/unified-api-client/unified-api-client';
+import { PrismaService } from '@/prisma/prisma.service';
+import { beginTransactionIfNotExistCurry } from '@/prisma/prisma.util';
+import { TCustomerConfig } from '@/customer/schemas/zod-schemas';
 
 @Injectable()
 export class BusinessService {
+  private readonly unifiedApiClient = new UnifiedApiClient();
+
   constructor(
     protected readonly repository: BusinessRepository,
     protected readonly logger: AppLoggerService,
     protected readonly httpService: HttpService,
+    protected readonly customerService: CustomerService,
+    private readonly prisma: PrismaService,
   ) {}
+
   async create(args: Parameters<BusinessRepository['create']>[0], transaction?: PrismaTransaction) {
-    return await this.repository.create(args, transaction);
+    return await beginTransactionIfNotExistCurry({
+      transaction,
+      prismaService: this.prisma,
+    })(async tx => {
+      const business = await this.repository.create(args, tx);
+
+      const businessPayload = (await this.repository.findByIdUnscoped(
+        business.id,
+        {
+          select: {
+            id: true,
+            correlationId: true,
+            companyName: true,
+            metadata: true,
+            createdAt: true,
+            updatedAt: true,
+            project: {
+              select: {
+                customer: {
+                  select: {
+                    id: true,
+                    config: true,
+                  },
+                },
+              },
+            },
+          },
+        },
+        tx,
+      )) as Business & {
+        project: {
+          customer: {
+            id: string;
+            config: TCustomerConfig | null;
+          };
+        };
+      };
+
+      if (businessPayload.project.customer.config?.isMerchantMonitoringEnabled) {
```

Comment:
Its better to extract this check to a function, the logic will probably change over time (deciding which busnieses we sync)

Reply:
Fixed